### PR TITLE
Sticky Web

### PR DESCRIPTION
**Pull request by Koekenpann**
Few notes:

- The Defiant and Competitive case is probably not the best solution as it doesn't cover all the switch-in cases like Intimidate.
- However there is no longer a need for an edge-case where a Court Changed Sticky Web doesn't activate Defiant or Competitive. I have verified this on cartridge which I can show on Discord.
- It should have (some) foundation for Mirror Armor, as it stores the user, users side and the existence of Sticky Web as Aero suggested on Discord.
- [Contrary raises its Speed.](https://i.imgur.com/B5FuZ1v.gif)
- Defog (both sides) and Rapid Spin (user's side) remove the Sticky Web from the field.

melonDS_20240801-195931210

melonDS_20240801-195950447

Also fixes:
- Heavy-Duty Boots didn't protect Flying-types/Levitate Pokémon from Stealth Rock (I think).
- If Rapid Spin fainted something, then the switched-in Pokémon triggered a stat drop it would re-run the RapidSpin battlecommand.
- If a Comatose Pokémon switches in on Toxic Spikes + another entry hazard, it would not be affected by any entry hazard.
- Seems like #308 was already fixed before these changes.

Hope I covered everything, if not let me know!

### CODE REVIEW

File: src/battle/battle_script_commands.c [hunk 1]
```diff
@@ -3184,7 +3187,43 @@ BOOL BtlCmd_RapidSpin(void *bw, struct BattleStruct *sp)
...
     IncrementBattleScriptPtr(sp, 1);
+    int adrs = read_battle_script_param(sp);
+    int side = IsClientEnemy(bw, sp->defence_client);
+
+    if (sp->side_condition[side] & SIDE_STATUS_STICKY_WEB) {
+        sp->oneSelfFlag[sp->attack_client].no_pressure_flag = 1;
+        IncrementBattleScriptPtr(sp, adrs);
+    } else {
+        sp->side_condition[side] |= SIDE_STATUS_STICKY_WEB;
+        sp->scw[side].stickyWebBattlerId = sp->attack_client;                      // For Mirror Armor in doubles
```
Comment (Aeliko) [inline src/battle/battle_script_commands.c:3224]: If the Pokémon with this Ability is affected by [Sticky Web](https://bulbapedia.bulbagarden.net/wiki/Sticky_Web_(move)), the effect is reflected back to the Pokémon which set it up. If Pokémon which set up Sticky Web is not on the field, no Pokémon have their Speed lowered.

`sp->attack_client` is the position of the active battlers, would need to somehow handle this in however way you think is good

File: data/battle_scripts/subscripts/subscript_0012_UPDATE_STAT_STAGE.s [hunk 2]
```diff
@@ -14,6 +14,7 @@ _010:
     Wait 
 
 _018:
+    CompareVarToValue OPCODE_FLAG_NOT, BSCRIPT_VAR_BATTLE_STATUS, BATTLE_STATUS_MOVE_SUCCESSFUL, _023
```
Comment (BluRosie) [inline data/battle_scripts/subscripts/subscript_0012_UPDATE_STAT_STAGE.s:17]: oh shucks, was rapid spin always raising speed?  that's funny
should really do rapid spin properly at some point 😂

File: src/battle/battle_script_commands.c (hunk 1 above)
Comment (BluRosie) [reply src/battle/battle_script_commands.c:3224]: wait so...  mirror armor switch in will put the sticky web back on the other side of the field *IF* the mon that set it up is still there?

Comment (Aeliko) [reply src/battle/battle_script_commands.c:3224]: The stat drop itself, not the hazard.

Comment (BluRosie) [reply src/battle/battle_script_commands.c:3224]: oh thank goodness
so
`sp->scw[side].stickyWebBattlerId` is default 0, client + 1 if active, and we clear it with the other battle flags in [`ClearBattleMonFlags`](https://github.com/BluRosie/hg-engine/blob/3129587bfb4f0b6b3461aaba3f0feb435c17b8dc/src/battle/battle_pokemon.c#L992) maybe?  default `MAX_CLIENT` and `client` if active?

Comment (Aeliko) [reply src/battle/battle_script_commands.c:3224]: Sounds good to me!

**Review by BluRosie — APPROVED** [review]

**Pull request merged**